src/components/ColorPickerInput.tsx

```diff
@@ -0,0 +1,88 @@
+import {
+  useState,
+  useEffect,
+  useRef,
+  MutableRefObject,
+  useLayoutEffect,
+} from "react";
+import { Box, useTheme } from "@mui/material";
+
+import { Color, ColorPicker } from "react-color-palette";
+import { useDebouncedCallback } from "use-debounce";
+
+const useResponsiveWidth = (): [
+  width: number,
+  setRef: MutableRefObject<HTMLElement | null>
+] => {
+  const ref = useRef(null);
+  const [width, setWidth] = useState(0);
+
+  useLayoutEffect(() => {
+    if (!ref || !ref.current) {
+      return;
+    }
+    const resizeObserver = new ResizeObserver((targets) => {
+      const { width: currentWidth } = targets[0].contentRect;
+      setWidth(currentWidth);
+    });
+
+    resizeObserver.observe(ref.current);
+
+    return () => {
+      resizeObserver.disconnect();
+    };
+  }, []);
+
+  return [width, ref];
+};
+
+export interface IColorPickerProps {
+  value: Color;
+  handleOnChangeComplete: (color: Color) => void;
+  disabled?: boolean;
+}
+
+export default function ColorPickerInput({
+  value,
+  handleOnChangeComplete,
```

Comment:
```suggestion
  onChangeComplete: ColorPickerProps['onChangeComplete'];
  disabled?: boolean;
}

export default function ColorPickerInput({
  value,
  onChangeComplete,
```

Pass the `onChangeComplete` prop to the `<ColorPicker>` component directly. You also need to import `ColorPickerProps` from react-color-palette.

Reply:
This is a nice trick I did not know it, thanks!

Reply:
I think it's better not to use this as onChangeComplete is optional in ColorPickerProps.

Reply:
> I think it's better not to use this as onChangeComplete is optional in ColorPickerProps.

You can use `NonNullable<ColorPickerProps['onChangeComplete']>` [(TypeScript docs)](https://www.typescriptlang.org/docs/handbook/utility-types.html#nonnullabletype)